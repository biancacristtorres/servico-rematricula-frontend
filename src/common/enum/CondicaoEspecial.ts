export enum CondicaoEspecial {
  TurmasEspecias = 1,
  CampusDiferentes = 2,
  TurnosDiferentes = 3,
}
