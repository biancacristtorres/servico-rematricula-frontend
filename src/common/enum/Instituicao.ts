export enum Instituicao {
  UniBhBeloHorizonte = 1,
  UnaBeloHorizonte = 3,
  Contagem = 4,
  Unimonte = 5,
  Betim = 6,
  Ustj = 8,
  SeteLagoas = 9,
  Divinopolis = 10,
  PousoAlegre = 11,
  Joinville = 12,
  Balneario = 13,
  Blumenau = 15,
  Florianopolis = 16,
  Curitiba = 17,
  SaoBentoDoSul = 18,
  BomDespacho = 19,
  Uberlandia = 20,
  Catalao = 21,
  NovaSerrana = 22,
  EssjGuarulhos = 23,
  FacsjGuarulhos = 24,
  IusjGuarulhos = 25,
  EssjLimeira = 26,
  FacsjLimeira = 27,
  IusjLimeira = 28,
  EssjSaoBernado = 29,
  FacsjSaoBernado = 30,
  IusjSaoBernado = 31,
}
