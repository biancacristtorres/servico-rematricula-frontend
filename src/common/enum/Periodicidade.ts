export enum Periodicidade {
  Semestral = 1,
  Anual = 2,
  Bimestral = 3,
  Trimestral = 4,
  Mensal = 5,
}
