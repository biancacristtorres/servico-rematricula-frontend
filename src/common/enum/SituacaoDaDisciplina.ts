export enum SituacaoDaDisciplina {
  Solicitada = 7,
  Confirmada = 5,
  Reservada = 2,
  Liberada = 1,
}
