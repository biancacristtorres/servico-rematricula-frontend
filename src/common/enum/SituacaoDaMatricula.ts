export enum SituacaoDaMatricula {
  Processando = 1,
  Liberada = 2,
}
