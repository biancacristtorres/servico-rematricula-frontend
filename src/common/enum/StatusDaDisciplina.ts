export enum StatusDaDisciplina {
  FilaDeEspera = 7,
  AguardandoConfirmacao = 2,
  Confirmada = 5,
}
