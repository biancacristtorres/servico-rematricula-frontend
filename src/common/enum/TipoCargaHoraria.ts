export enum TipoCargaHoraria  {
  Teorica = 1,
  Pratica = 2,
}
