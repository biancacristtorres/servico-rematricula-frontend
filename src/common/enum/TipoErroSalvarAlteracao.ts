export enum TipoErroSalvarAlteracao {
  MINIMO_1_DISCIPLINAS = 'MINIMO_1_DISCIPLINAS',
  MINIMO_2_DISCIPLINAS = 'MINIMO_2_DISCIPLINAS',
  MINIMO_3_DISCIPLINAS = 'MINIMO_3_DISCIPLINAS',
  QUEBRA_MODULO = 'QUEBRA_MODULO',
}
