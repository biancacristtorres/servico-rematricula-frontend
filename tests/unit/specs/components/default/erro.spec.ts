import { ErroFactory } from '../../../factory/components/ErroFactory';
import { Instituicao } from '@/common/enum/Instituicao';

describe('Componente de Erro', () => {
  const erroFactory = new ErroFactory();

  beforeEach(() => {
    erroFactory.dadoQueOAlunoSejaDaInstuicaoUNA();
  });

  it('O título exibido deve ser o título passado por parâmetro', async (done) => {
    erroFactory.dadoOTitulo('XPTO');

    await erroFactory.montarComponente();

    expect(erroFactory.tituloDoErro).toEqual('XPTO');

    done();
  });

  it('O texto exibido deve ser o texto passado por parâmetro', async (done) => {
    erroFactory.dadoOTexto('XPTO');

    await erroFactory.montarComponente();

    expect(erroFactory.TextoDoErro).toEqual('XPTO');

    done();
  });

  it('Caso um nome válido de imagem seja passado por parâmetro a imagem deve ser renderizada', async (done) => {
    erroFactory.dadoAImagem('sistema-indisponivel');

    await erroFactory.montarComponente();

    expect(erroFactory.existeAImagem).toBeTruthy();

    done();
  });

  it('O botão de redirecionamento para o SOL deve aparecer', async (done) => {
    erroFactory.dadoAExibicaoDoBotaoDeRerecionamento(true);

    await erroFactory.montarComponente();

    expect(erroFactory.retornarAoSol.exists()).toBeTruthy();
    done();
  });

  it('O botão de redirecionamento para o SOL não deve aparecer', async (done) => {
    erroFactory.dadoAExibicaoDoBotaoDeRerecionamento(false);

    await erroFactory.montarComponente();

    expect(erroFactory.retornarAoSol.exists()).toBeFalsy();
    done();
  });

  it('Possui um botão para redirecionar para o SOL com uma url dinâmica, com base na instituição', async (done) => {
    erroFactory.dadoAExibicaoDoBotaoDeRerecionamento(true);

    await erroFactory.montarComponente();
    const atual = erroFactory.retornarAoSol.attributes('href');

    const esperado = `https://aluno.una.br/SOL/aluno/index.php/index/seguranca/dev/instituicao/${
      Instituicao.UnaBeloHorizonte
    }`;
    expect(atual).toEqual(esperado);
    done();
  });

  it('Snapshot', async (done) => {
    erroFactory.dadoAExibicaoDoBotaoDeRerecionamento(true);

    await erroFactory.montarComponente();

    expect(erroFactory.componente).toMatchSnapshot();

    done();
  });
});
