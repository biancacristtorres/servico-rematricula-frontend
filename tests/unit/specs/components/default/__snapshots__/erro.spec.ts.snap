// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente de Erro Snapshot 1`] = `
<vcontainer-stub tag="div" fluid="" class="grey--text">
  <vlayout-stub tag="div" wrap="" align-center="" row="" reverse="" justify-center="" fill-height="">
    <vflex-stub tag="div" xs12="" md6="" class="text-xs-center pt-5">
      <vlayout-stub tag="div" wrap="" row="" fill-height="" justify-start="true">
        <vflex-stub tag="div" xs12="" class="coluna">
          <vimg-stub src="[object Object]" position="center center" transition="fade-transition" id="imagem-erro" class="text-xs-center coluna"></vimg-stub>
        </vflex-stub>
      </vlayout-stub>
    </vflex-stub>
    <vflex-stub tag="div" xs12="" md6="">
      <vlayout-stub tag="div" wrap="" row="" fill-height="" justify-end="true">
        <vflex-stub tag="div" xs12="" class="coluna">
          <p id="titulo-erro" class="display-2 text-xs-left">
            XPTO
          </p>
        </vflex-stub>
        <vflex-stub tag="div" xs12="" class="coluna">
          <div id="texto-erro" class="pt-3 texto subheading text-xs-left">
            XPTO
          </div>
        </vflex-stub>
        <vflex-stub tag="div" xs12="" class="coluna">
          <vbtn-stub activeclass="v-btn--active" tag="button" type="button" id="retornar-ao-sol" href="https://aluno.una.br/SOL/aluno/index.php/index/seguranca/dev/instituicao/3" color="primary" class="mt-3 ml-0 btn-retornar-ao-sol">
            Retornar ao SOL
          </vbtn-stub>
        </vflex-stub>
      </vlayout-stub>
    </vflex-stub>
  </vlayout-stub>
</vcontainer-stub>
`;
